src/types.rs

```diff
@@ -246,6 +247,13 @@ impl Type {
         Self(TypeEnum::Sum(SumType::Simple { size }), TypeBound::Eq)
     }
 
+    /// New type variable (for use in type schemes only),
+    /// with bound matching that in the type scheme
+    /// (i.e. the variable must be declared as a [TypeParam::Type])
+    pub fn new_variable(idx: usize, bound: TypeBound) -> Self {
```

Comment:
Hmmm, this should possibly be called `use_var` (a) for consistency with TypeArg::use_var, and (b) to emphasize that it doesn't *create* (i.e. declare) a new variable, merely a new reference/usage of an existing variable (declared elsewhere). `new_variable` may be consistent with `Type::new_extension`, `Type::new_function` etc., but is potentially misleading.

Reply:
I've gone with `new_var_use` in both, hope that makes it clear